userspace/libsinsp/utils.cpp

```diff
@@ -1904,3 +1904,101 @@ void sinsp_scap_debug_log_fn(const char* msg)
 {
 	g_logger.log(msg, sinsp_logger::SEV_DEBUG);
 }
+
+///////////////////////////////////////////////////////////////////////////////
+// Set operation functions.
+///////////////////////////////////////////////////////////////////////////////
+
+// unordered_set_to_ordered
+template<typename T>
+std::set<T> unordered_set_to_ordered(std::unordered_set<T> unordered_set)
+{
+	std::set<T> s;
+	for(const auto& val : unordered_set)
+	{
+		s.insert(val);
+	}
+	return s;
+}
+template std::set<uint32_t> unordered_set_to_ordered(std::unordered_set<uint32_t> unordered_set);
+template std::set<std::string> unordered_set_to_ordered(std::unordered_set<std::string> unordered_set);
+
+// convert_vector_to_unordered_set
+template<typename T>
+std::unordered_set<T> convert_vector_to_unordered_set(std::vector<T> v)
+{
+	std::unordered_set<T> s;
+	for (const auto &x : v)
+	{
+		s.insert(x);
+	}
+	return s;
+}
+template std::unordered_set<std::string> convert_vector_to_unordered_set(std::vector<std::string> v);
+template std::unordered_set<uint32_t> convert_vector_to_unordered_set(std::vector<uint32_t> v);
+
+// unordered_set_difference, equivalent to SQL left_anti join operation
+template<typename T>
+std::unordered_set<T> unordered_set_difference(std::unordered_set<T> a, std::unordered_set<T> b)
+{
+	std::set<T> c = unordered_set_to_ordered(a);
+	std::set<T> d = unordered_set_to_ordered(b);
+	std::vector<T> v;
+	std::set_difference(c.begin(), c.end(), d.begin(), d.end(), std::inserter(v, v.begin()));
+	return convert_vector_to_unordered_set(v);
```

Comment:
```suggestion
	std::unordered_set<T> s;
	for(const auto& val : a)
	{
		if (b.find(val) == b.end())
		{
			s.insert(val);
		}
	}
	return s;
```
I think this is a simpler and generally more performant implementation. We don't know where these functions will be used in the future, but considering all the performance-critical paths we have I think that an extra care on performance is always good.